src/DotNet/Writer/Metadata.cs

```diff
@@ -2704,7 +2704,15 @@ struct MethodScopeDebugInfo {
 				if (!VerifyFieldSize(field, ivBytes.Length))
 					Error("Field '{0}' (0x{1:X8}) initial value size != size of field type.", field, field.MDToken.Raw);
 				uint rid = GetRid(field);
-				var iv = constants.Add(new ByteArrayChunk(ivBytes), ModuleWriterBase.DEFAULT_CONSTANTS_ALIGNMENT);
+
+				uint alignment = ModuleWriterBase.DEFAULT_CONSTANTS_ALIGNMENT;
+				const uint MaxFieldInitialValueAlignment = 1024U;
+				if (field.FieldType is TypeDefOrRefSig tdrSig && tdrSig.TypeDef?.ClassLayout is {} classLayout) {
+					uint requiredAlignment = Math.Min(Utils.RoundToNextPowerOfTwo(classLayout.PackingSize), MaxFieldInitialValueAlignment);
```

Comment:
We should produce an error instead of ignoring too big alignment values.

Or since PackingSize is just a 16-bit integer, we might as well just allow any 16-bit value.